metadata-ingestion/tests/integration/kafka-connect/test_kafka_connect.py

```diff
@@ -252,3 +253,76 @@ def test_kafka_connect_ingest(docker_compose_runner, pytestconfig, tmp_path, moc
         )
+
+
+@freeze_time(FROZEN_TIME)
+@pytest.mark.integration_batch_1
+def test_kafka_connect_mongosourceconnect_ingest(
+    docker_compose_runner, pytestconfig, tmp_path, mock_time
+):
+    test_resources_dir = pytestconfig.rootpath / "tests/integration/kafka-connect"
+    test_resources_dir_kafka = pytestconfig.rootpath / "tests/integration/kafka"
+
+    # Share Compose configurations between files and projects
+    # https://docs.docker.com/compose/extends/
+    docker_compose_file = [
+        str(test_resources_dir_kafka / "docker-compose.yml"),
+        str(test_resources_dir / "docker-compose.override.yml"),
+    ]
+    with docker_compose_runner(docker_compose_file, "kafka-connect") as docker_services:
+        wait_for_port(docker_services, "test_broker", 59092, timeout=120)
+        wait_for_port(docker_services, "test_connect", 58083, timeout=120)
+        docker_services.wait_until_responsive(
+            timeout=30,
+            pause=1,
+            check=lambda: requests.get(
+                "http://localhost:58083/connectors",
+            ).status_code
+            == 200,
+        )
+        # Creating MongoDB source
+        r = requests.post(
+            "http://localhost:58083/connectors",
+            headers={"Content-Type": "application/json"},
+            data=r"""{
+                    "name": "source_mongodb_connector",
+                    "config": {
+                        "tasks.max": "1",
+                        "connector.class": "com.mongodb.kafka.connect.MongoSourceConnector",
+                        "connection.uri": "mongodb://admin:admin@test_mongo:27017",
+                        "topic.prefix": "mongodb",
+                        "database": "test_db",
+                        "collection": "purchases",
+                        "copy.existing": true,
+                        "copy.existing.namespace.regex": "test_db.purchases",
+                        "change.stream.full.document": "updateLookup",
+                        "topic.creation.enable": "true",
+                        "topic.creation.default.replication.factor": "-1",
+                        "topic.creation.default.partitions": "-1",
+                        "output.json.formatter": "com.mongodb.kafka.connect.source.json.formatter.SimplifiedJson",
+                        "key.converter": "org.apache.kafka.connect.storage.StringConverter",
+                        "value.converter": "org.apache.kafka.connect.storage.StringConverter",
+                        "key.converter.schemas.enable": false,
+                        "value.converter.schemas.enable": false,
+                        "output.format.key": "schema",
+                        "output.format.value": "json",
+                        "output.schema.infer.value": false,
+                        "publish.full.document.only":true
+                    }
+                }""",
+        )
+
+        # Give time for connectors to process the table data
```

Comment:
Consider adding below line to verify that connector got created successfully.
`assert r.status_code == 201  # Created`